src/Orleans/Messaging/Message.cs

```diff
@@ -440,24 +444,15 @@ public Message()
             headerBytes = null;
         }
 
-        private Message(Categories type, Directions subtype)
-            : this()
-        {
-            Category = type;
-            Direction = subtype;
-        }
-
         internal static Message CreateMessage(InvokeMethodRequest request, InvokeMethodOptions options)
         {
-            var message = new Message(
```

Comment:
This PR is a good change, but it will make it quite easy for maintainers, without your level of insight, to introduce changes that could leak messages.  Having some means of detecting leaks would be valuable.

Maybe add a static int messageCount that we increment every allocation, then log that in the metrics.   If we're logging this count in the metrics, over time, the count should become a constant.  If it's constantly increasing, it would indicate a leak.  Other methods of leak detection are welcome, but I think we need some for of detection.